[New Product] Plesk

sibprogrammer opened the pull request.

sibprogrammer committed:
Add Plesk product

marcwrobel commented on products/plesk.md:
```diff
@@ -0,0 +1,35 @@
+    link: https://docs.plesk.com/release-notes/onyx/change-log/
+---
+
+> [Plesk](https://www.plesk.com/) is a proprietary hosting control panel developed by WebPros
```
A link to https://webpros.com/ would be great too.

sibprogrammer replied:
Added.

marcwrobel commented on products/plesk.md:
```diff
@@ -0,0 +1,35 @@
+permalink: /plesk
+versionCommand: plesk version
+releasePolicyLink: https://www.plesk.com/lifecycle-policy/
+eolColumn: End Of Life
```
The label `Support` would be more in line with what's already done on other pages.

sibprogrammer replied:
Changed.

marcwrobel commented on products/plesk.md:
```diff
@@ -0,0 +1,35 @@
+releaseDateColumn: true
+
+releases:
+-   releaseCycle: "18.0.50"
```
My first thought was to say that the cycle should be only `18`. But given the new release policy adopted for Plesk 18.x I understand why you made such choice.

https://docs.plesk.com/release-notes/obsidian/change-log seems to list all Plesk 18 releases with their date, so it should be possible to write a custom automation script that would :

- update automatically tiny releases (such as 18.0.48.1, 18.0.48.2...),
- let us know when there are new releases (such as 18.0.51, 18.0.52...).

Unfortunately we can't add new release cycles automatically for the time being, we can only update the existing one.

sibprogrammer replied:
>  it should be possible to write a custom automation script 

I can even help to write it, but I'm not very familiar with the project yet. If you have the examples, please point out on them.

> we can't add new release cycles automatically

I'm not sure how to address this. In one hand, we support two releases simultaneously, so I want to highlight the fact that not only the last release is supported. On the other hand, we can keep the information only about latest release in this MD file (cons: no history, no info about one more supported version of the product).

marcwrobel replied:
> I can even help to write it, but I'm not very familiar with the project yet. If you have the examples, please point out on them.

Scripts are located in https://github.com/endoflife-date/release-data. Every 6 hours the scripts are ran to retrieve new releases an store them as JSON in https://github.com/endoflife-date/release-data/tree/main/releases.

And every day, the releases in product's markdown files are updated using the data stored in the JSON files. That's done in [latest.py](https://github.com/endoflife-date/endoflife.date/blob/master/_auto/latest.py).

I think the script for plesk should be similar to what's done in [cos.py](https://github.com/endoflife-date/release-data/blob/main/src/cos.py).

marcwrobel replied:
> I'm not sure how to address this. In one hand, we support two releases simultaneously, so I want to highlight the fact that not only the last release is supported. On the other hand, we can keep the information only about latest release in this MD file (cons: no history, no info about one more supported version of the product).

There is no need to address this one and we can keep the current approach. New releases will be detected every month (you can take a look at https://github.com/endoflife-date/endoflife.date/pull/2550#issuecomment-1438442065 for an example). And someone will have to manually add the new release in the markdown front matter.

marcwrobel replied:
Automation will be implemented in https://github.com/endoflife-date/release-data/issues/78.

marcwrobel commented on products/plesk.md:
```diff
@@ -0,0 +1,35 @@
+    link: https://docs.plesk.com/release-notes/onyx/change-log/
+---
+
+> [Plesk](https://www.plesk.com/) is a proprietary hosting control panel developed by WebPros
```
A summary of the "Plesk Lifecycle Policy" page would be great. Something like :

> In the past (e.g. 12.x > 17.x), Plesk used to launch one major version per year. With Plesk 18.x the release cadence changed to one patch release a month, with each release supported for one two months...

sibprogrammer replied:
I prepared a recap of the policy. Hope it helps.

marcwrobel replied:
Yep. I updated it a little bit, let me know if it looks good to you.

A preview is available at https://deploy-preview-2544--endoflife-date.netlify.app/plesk.

sibprogrammer replied:
Thank you, looks very good! 👍

marcwrobel reviewed (changes requested):
@sibprogrammer, thank you for the PR and sorry again for my first comment (deleted now). Here are some notes and suggestions.

marcwrobel committed:
[plesk] Use changelogTemplate

marcwrobel committed:
[plesk] Improve description

Using https://docs.plesk.com/en-US/obsidian/administrator-guide/about-plesk.70559/

marcwrobel committed:
[plesk] Update release policy

marcwrobel reviewed (approved):
LGTM, merging.
